lib/routes/veeam/put.js

```diff
@@ -0,0 +1,90 @@
+
+const async = require('async');
+const { parseString } = require('xml2js');
+const { receiveData } = require('./utils');
+const { s3routes, errors } = require('arsenal');
+const metadata = require('../../metadata/wrapper');
+const parseSystemSchema = require('./schemas/system');
+const parseCapacitySchema = require('./schemas/capacity');
+const writeContinue = require('../../utilities/writeContinue');
+
+const { responseNoBody, responseXMLBody } = s3routes.routesUtils;
+
+/**
+ * Puts a veeam capacity or system file in the bucket metadata.
+ * Logic ensures consistency of the data and metadata.
+ *
+ * @param {object} request - request object
+ * @param {object} response - response object
+ * @param {object} bucketMd - bucket metadata from the db
+ * @param {object} log - logger object
+ * @returns {undefined} -
+ */
+function putVeeamFile(request, response, bucketMd, log) {
+    if (!bucketMd) {
+        return errors.NoSuchBucket;
+    }
+
+    return async.waterfall([
+        next => {
+            // Extract the data from the request, keep it in memory
+            writeContinue(request, response);
+            return receiveData(request, log, next);
+        },
+        (value, next) => parseString(value, { explicitArray: false }, (err, parsed) => {
+            // Convert the received XML to a JS object
+            if (err) {
+                return next(errors.MalformedXML);
+            }
+            return next(null, parsed);
+        }),
+        (parsedXML, next) => {
+            const isSystemXML = request.objectKey.endsWith('/system.xml');
+            const capabilities = bucketMd._capabilities || {
+                VeeamSOSApi: {},
+            };
+            // Initialize Veeam capability if not already there
+            if (!capabilities.VeeamSOSApi) {
+                capabilities.VeeamSOSApi = {};
+            }
+            // Validate the JS object schema with joi and prepare the object for
+            // further logic
+            const validateFn = isSystemXML ? parseSystemSchema : parseCapacitySchema;
+            let validatedData = null;
+            try {
+                validatedData = validateFn(parsedXML);
+            } catch (err) {
+                log.error('xml file did not pass validation', { err });
+                return next(errors.MalformedXML);
+            }
+            // System and Capacity files also include the last-modified date used by Veeam
+            const lastModified = (new Date()).toISOString();
+            if (isSystemXML) {
```

Comment:
Done with error handling: https://github.com/scality/cloudserver/pull/4942/commits/8c671ef0ac52fb1f0100cbc68a7f5da0eb9388d5#diff-38a3a2f10ad2b2920ef0e2d5815a184a4bdaab48fe7c82ecc0767c0be2fb8c8aR58

Also, I cannot set to capabilities directly, it would override existing data, so I'm just overriding if needed.